database/migrations/000035_provider_class_nullable.up.sql

```diff
@@ -0,0 +1,16 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+ALTER TABLE providers
+  ALTER COLUMN class DROP NOT NULL;
```

Comment:
What would a Null value mean here?

Reply:
a classless provider. That is, you'd have to specify all the `implements` and `authorization_flows` manually instead of using a class configuration.